node/src/actors/chain_manager/handlers.rs

```diff
@@ -1301,12 +1301,23 @@ impl Handler<GetReputation> for ChainManager {
             None => return Err(ChainManagerError::ChainNotReady.into()),
         };
 
-        Ok((rep_eng.trs().get(&pkh), rep_eng.ars().contains(&pkh)))
+        let reputation = rep_eng.trs().get(&pkh);
+        let eligibility = rep_eng.get_eligibility(&pkh) + 1;
+        let is_active = rep_eng.ars().contains(&pkh);
+        let total_reputation = rep_eng.total_active_reputation();
+
+        let rep_stats = ReputationStats {
+            reputation,
+            eligibility,
+            is_active,
+        };
+
+        Ok((rep_stats, total_reputation))
     }
 }
 
 impl Handler<GetReputationAll> for ChainManager {
-    type Result = Result<HashMap<PublicKeyHash, (Reputation, bool)>, failure::Error>;
+    type Result = Result<(u64, HashMap<PublicKeyHash, ReputationStats>), failure::Error>;
```

Comment:
`GetReputation` returns `(ReputationStats, u64)`, but `GetReputationAll` returns `(u64, HashMap)`. This is a bit confusing, so instead how about explicitly stating the result, like

```rust
struct GetReputationResult {
    stats: HashMap<PublicKeyHash, ReputationStats>,
    total_reputation: u64,
}
```

And we could unify the return types of `GetReputation` and `GetReputationAll`, `GetReputation` would return a `stats` with one element.